tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,339 @@
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates users to be reused across tests"""
+        super(HammerAuthTestCase, cls).setUpClass()
+        cls.uname_admin = gen_string('alpha')
+        cls.uname_viewer = gen_string('alpha')
+        cls.password = gen_string('alpha')
+        cls.mail = 'test@example.com'
+        User.create({
+                'auth-source-id': '1',
+                'login': cls.uname_admin,
+                'password': cls.password,
+                'mail': cls.mail,
+                'admin': '1',
+        })
+        User.create({
+                'auth-source-id': '1',
+                'login': cls.uname_viewer,
+                'mail': cls.mail,
+                'password': cls.password,
+        })
+        User.add_role({'login': cls.uname_viewer, 'role': 'Viewer'})
+
+    @tier1
+    def test_positive_create_session(self):
+        """Check if user stays authenticated with session enabled
+
+        :id: fcee7f5f-1040-41a9-bf17-6d0c24a93e22
+
+        :Steps:
+
+            1. Set use_sessions, set short expiration time
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Wait until session expires, assert credentials
+               are required
+
+        :expectedresults: The session is successfully created and
+            expires after specified time
+        """
+        try:
+            configure_sessions()
+            Settings.set({'name': 'idle_timeout', 'value': 1})
+            Auth.login({
+                'username': self.uname_admin,
+                'password': self.password
+            })
+            result = Auth.status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            # list organizations without supplying credentials
+            orglist = Org.list(credentials=False, return_raw_response=True)
+            self.assertEqual(orglist.return_code, 0)
+            # wait until session expires
+            sleep(70)
+            orglist = Org.list(credentials=False, return_raw_response=True)
+            self.assertNotEqual(orglist.return_code, 0)
+            result = Auth.status()
+            self.assertIn(
+                LOGEDOFF_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+        finally:
+            # reset timeout to default
+            Settings.set({'name': 'idle_timeout', 'value': 60})
+
+    @tier1
+    def test_positive_disable_session(self):
+        """Check if user logs out when session is disabled
+
+        :id: 38ee0d85-c2fe-4cac-a992-c5dbcec11031
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Disable use_sessions
+
+        :expectedresults: The session is terminated
+
+        """
+        configure_sessions()
```

Comment:
Have the file created by this helper any impact on other tests? I mean, won't its presence interfere with tests which use common authentication? If there's any, even smallest chance that it will affect the tests - maybe it's worth deleting the file in tearDownClass just in case?